File: app/typescript-4/src/score.ts
export class Score {
  private score: number = 0;

  getScore(): number {
    return this.score;
  }

  addScore(points: number): void {
    this.score += points;
  }

  addZenkeshiBonus(): void {
    // 全消しボーナス（固定値）
    const zenkeshiBonus = 3600;
    this.score += zenkeshiBonus;
  }

  reset(): void {
    this.score = 0;
  }
}

File: app/fsharp-2/wpf/src/PuyoPuyo.WPF/Domain/Score.fs
namespace PuyoPuyo.Domain

/// スコアを表す型
type Score = { Value: int }

module Score =
    /// スコアを作成
    let create () : Score = { Value = 0 }

    /// スコアを加算
    let addScore (points: int) (score: Score) : Score = { score with Value = score.Value + points }

    /// 全消しボーナスを加算
    let addZenkeshiBonus (score: Score) : Score =
        let zenkeshiBonus = 3600
        addScore zenkeshiBonus score
